src/components/Questionnaire/QuestionTypes/SymptomQuestion.tsx

```diff
@@ -531,11 +619,15 @@ export function SymptomQuestion({
 
     if (isDuplicate) {
       toast.warning(t("symptom_already_exist_warning"));
+      setShowSymptomSelection(false);
       return;
     }
 
-    setSelectedCode(code);
-    setNewSymptom((prev) => ({ ...prev, code }));
+    setNewSymptom({
+      ...SYMPTOM_INITIAL_VALUE,
+      onset: { onset_datetime: new Date().toISOString().split("T")[0] },
+      code,
+    });
```

Comment:
this would cause weird issues with timezone! i remember having to revert this change of using `toISOString().split...` when building scheduling!

use `dateQueryString` util instead, or directly format to expected format using date utility library.

We should not be using this `split` on Date's ISO string ideally in the entire codebase!

Date would actually be different than intended on many edge cases.